### PR TITLE
fix: avoid leak of tenant token on the logs

### PR DESCRIPTION
**Pull request by MuchoLucho**
Changelog: None
Ticket: None

# External Contributor Checklist

🚨 Please review the [guidelines for contributing](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md) to this repository.

- [x] Make sure that all commits follow the conventional commit [specification](https://www.github.com/mendersoftware/mendertesting/commitlint/grammar.md) for the Mender project.

The majority of our contributions are fixes, which means your commit should have
the form below:

```
fix: <SHORT DESCRIPTION OF FIX>

<OPTIONAL LONGER DESCRIPTION>

Changelog: <USER-FRIENDLY-CHANGE-DESCRIPTION> or <None>
Ticket: <TICKET NUMBER> or <None>
```

- [x] Make sure that all commits are signed with [`git --signoff`](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). Also note that the signoff author must match the author of the commit.

### Description

Please describe your pull request.

Thank you!

### CODE REVIEW

**Comment by MuchoLucho** [comment]
I am not a Go guy, so I am sure it can be made more elegant. Open to suggestions

**Comment by buentead** [comment]
@MuchoLucho : Thanks for the change. From a security point of view a must have. It will be easier to share the debug logs. Very much appreciated!

File: app/auth.go [hunk 1]
```diff
@@ -676,7 +676,14 @@ func (m *menderAuthManagerService) MakeAuthRequest() (*client.AuthRequest, error
...
 
-	log.Debugf("Tenant token: %s", tentok)
+	tentok4Log := tentok
+
+	// Truncates the tenant token for avoid any leak
+	if len(tentok) > 4 {
+		tentok4Log = tentok[0:2] + "..." + tentok[len(tentok)-2:]
+	}
+
+	log.Debugf("Tenant token: %s", tentok4Log)
```
Comment (lluiscampos) [inline app/auth.go:686]: With regards to the general logic. It won't happen in practice, but look at the following:
* do a copy of tenok to tenok4log
* `if` something something, then redact tenok4log for sercure logging
* log unconditionally

See the issue? If the `if` is false then you log the original secret tenok.

There are many ways to solve this. One way:
```suggestion
	tentok4Log := ""

	// Truncates the tenant token for avoid any leak
	if len(tentok) > 4 {
		tentok4Log = tentok[0:2] + "..." + tentok[len(tentok)-2:]
	}

	log.Debugf("Tenant token: %s", tentok4Log)
```

and a simpler one:

```suggestion
	// Truncates the tenant token for avoid any leak
	if len(tentok) > 4 {
		log.Debugf("Tenant token: %s..%s", tentok[0:2], tentok[len(tentok)-2:])
	}
```

File: app/auth.go [hunk 2]
```diff
@@ -676,7 +676,14 @@ func (m *menderAuthManagerService) MakeAuthRequest() (*client.AuthRequest, error
...
 
 	// fill tenant token
 	authd.TenantToken = string(tentok)
```
Comment (lluiscampos) [inline app/auth.go:689]: Two lines below we have `log.Debugf("Authorization data: %v", authd)` which also will leak the token, right?

**Review by lluiscampos — CHANGES_REQUESTED** [review]
Great first start in golang, @MuchoLucho ! See my comments below I think we can brush this up and merge it.

Also please add a good changelog entry, this is clearly user facing and we will release it.

**Comment by lluiscampos** [comment]
3.5.x is EOL, closing this PR.

Thank you @MuchoLucho regardless for your efforts :smiley:

**Pull request closed without merging**